recipes/stb/all/conanfile.py

```diff
@@ -1,27 +1,53 @@
 from conans import ConanFile, tools
 import os
 
+required_conan_version = ">=1.33.0"
+
+
 class StbConan(ConanFile):
     name = "stb"
     description = "single-file public domain libraries for C/C++"
-    topics = ("conan", "stb", "single-file")
+    topics = ("stb", "single-file")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/nothings/stb"
     license = ("Unlicense", "MIT")
     no_copy_source = True
-    _source_subfolder = "source_subfolder"
+
+    options = {
+        "with_deprecated": [True, False]
+    }
+
+    default_options = {
+        "with_deprecated": True
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _version(self):
+        # HACK: Used to circumvent the incompatibility
+        #       of the format cci.YYYYMMDD in tools.Version
+        return str(self.version)[4:]
+
+    def config_options(self):
+        if tools.Version(self._version) < "20210713":
+            del self.options.with_deprecated
 
     def source(self):
-        commit = os.path.splitext(os.path.basename(self.conan_data["sources"][self.version]["url"]))[0]
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + commit
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 
     def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         self.copy("*.h", src=self._source_subfolder, dst="include")
         self.copy("stb_vorbis.c", src=self._source_subfolder, dst="include")
         tools.rmdir(os.path.join(self.package_folder, "include", "tests"))
+        if tools.Version(self._version) >= "20210713":
+            tools.rmdir(os.path.join(self.package_folder, "include", "deprecated"))
```

Comment:
I think you can do this unconditionally.
It won't fail if the directory does not exist.

Reply:
Personally, I prefer to have things like this directly connected to recipe version. Helps me to understand immediately what behaviour / files to except in specific versions

Reply:
I did this with the "Better safe than sorry" mentality, but also wanting to explicitly express differences between the versions.